Android/src/org/droidplanner/android/maps/providers/baidu_map/BaiduMapOfflineMapPreference.java

```diff
@@ -0,0 +1,357 @@
+        Toast.makeText(context, text, Toast.LENGTH_SHORT).show();
+        updateView();
+    }
+
+    /**
+     * Display offline map update status
+     */
+    public void updateView() {
+        mLocalMapList = mOffline.getAllUpdateInfo();
+        if (mLocalMapList == null) {
+            mLocalMapList = new ArrayList<MKOLUpdateElement>();
+        }
+        mLAdapter.notifyDataSetChanged();
+    }
+
+    public String formatDataSize(int size) {
+        String ret = "";
+        if (size < (DATA_SIZE_ONE_KB)) {
+            ret = String.format("%dK", size / DATA_SIZE_UNIT_KB);
+        } else {
+            ret = String.format("%.1fM", size / (DATA_SIZE_UNIT_MB));
+        }
+        return ret;
+    }
+
+    @Override
+    protected void onDialogClosed(boolean positiveResult) {
+        // destory offline module when exit
+        mOffline.destroy();
+        super.onDialogClosed(positiveResult);
+    }
+
+
+    @Override
+    public void onBindDialogView(View view){
+        // Initialize Baidu Map library
+        com.baidu.mapapi.SDKInitializer.initialize(this.getContext().getApplicationContext());
+        mBindDialogView = view;
+        mOffline = new MKOfflineMap();
+        mOffline.init(this);
+
+        initView(view);
+        super.onBindDialogView(view);
+    }
+
+    @Override
+    public void onGetOfflineMapState(int type, int state) {
+        switch (type) {
+            case MKOfflineMap.TYPE_DOWNLOAD_UPDATE: {
+                MKOLUpdateElement update = mOffline.getUpdateInfo(state);
+                // display download progress
+                if (update != null) {
+                    mStateView.setText(String.format("%s : %d%%", update.cityName, update.ratio));
+                    updateView();
+                }
+            }
+            break;
+            case MKOfflineMap.TYPE_NEW_OFFLINE:
+                // Prompt offline map update is available
+                Log.d("Baidu offline map", String.format("add offlinemap num:%d", state));
+                break;
+            case MKOfflineMap.TYPE_VER_UPDATE:
+                // Not to prompt version update
+                break;
+            default:
+                break;
+        }
+    }
+
+    /**
+     *  list adapter for managing offline map
+     */
+    public class LocalMapAdapter extends BaseAdapter {
+
+        @Override
+        public int getCount() {
+            return mLocalMapList.size();
+        }
+
+        @Override
+        public Object getItem(int index) {
+            return mLocalMapList.get(index);
+        }
+
+        @Override
+        public long getItemId(int index) {
+            return index;
+        }
+
+        @Override
+        public View getView(int index, View view, ViewGroup arg2) {
+            MKOLUpdateElement e = (MKOLUpdateElement) getItem(index);
+            view = View.inflate(BaiduMapOfflineMapPreference.this.getContext(), R.layout.list_baidumap_offline_localmap_item, null);
```

Comment:
@droneboost if the `view` parameter is non null, you can reuse it instead of inflating a new one. This helps with performance since view inflation is expansive. You can see an example [here](https://github.com/DroidPlanner/Tower/blob/a567a195b3cb9a815fb080204a4130e618753f75/Android/src/org/droidplanner/android/view/adapterViews/ParamsAdapter.java#L79)